<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>973</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Octree Testing</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QWidget" name="contentWidget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="SceneWidget" name="sceneWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>600</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="controlWidget" native="true">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QWidget" name="jointAngleControlWidget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="label_9">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <pointsize>12</pointsize>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>Joint Angle Controls (in dg)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="widget_4" native="true">
              <property name="font">
               <font>
                <pointsize>10</pointsize>
                <italic>true</italic>
               </font>
              </property>
              <layout class="QGridLayout" name="gridLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>J0 Angle</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="j0Field">
                 <property name="minimum">
                  <number>-720</number>
                 </property>
                 <property name="maximum">
                  <number>720</number>
                 </property>
                 <property name="singleStep">
                  <number>5</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>J1 Angle</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="j1Field">
                 <property name="minimum">
                  <number>-720</number>
                 </property>
                 <property name="maximum">
                  <number>720</number>
                 </property>
                 <property name="singleStep">
                  <number>5</number>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>J2 Angle</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QSpinBox" name="j2Field">
                 <property name="minimum">
                  <number>-720</number>
                 </property>
                 <property name="maximum">
                  <number>720</number>
                 </property>
                 <property name="singleStep">
                  <number>5</number>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>J3 Angle</string>
                 </property>
                </widget>
               </item>
               <item row="6" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>J5 Angle</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>J4 Angle</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QSpinBox" name="j3Field">
                 <property name="minimum">
                  <number>-720</number>
                 </property>
                 <property name="maximum">
                  <number>720</number>
                 </property>
                 <property name="singleStep">
                  <number>5</number>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QSpinBox" name="j4Field">
                 <property name="minimum">
                  <number>-720</number>
                 </property>
                 <property name="maximum">
                  <number>720</number>
                 </property>
                 <property name="singleStep">
                  <number>5</number>
                 </property>
                </widget>
               </item>
               <item row="6" column="1">
                <widget class="QSpinBox" name="j5Field">
                 <property name="minimum">
                  <number>-720</number>
                 </property>
                 <property name="maximum">
                  <number>720</number>
                 </property>
                 <property name="singleStep">
                  <number>5</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="jointPositionControlWidget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QLabel" name="label_10">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <pointsize>12</pointsize>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>Joint Position Controls</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="widget_5" native="true">
              <property name="font">
               <font>
                <pointsize>10</pointsize>
                <italic>true</italic>
               </font>
              </property>
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="4" column="1">
                <widget class="QDoubleSpinBox" name="zField">
                 <property name="minimum">
                  <double>-5.000000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>5.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.100000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="6" column="0">
                <widget class="QPushButton" name="decYButton">
                 <property name="text">
                  <string>Rotate Y-</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="1">
                <widget class="QPushButton" name="incXButton">
                 <property name="text">
                  <string>Rotate X+</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QDoubleSpinBox" name="yField">
                 <property name="minimum">
                  <double>-5.000000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>5.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.100000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="xLabel">
                 <property name="text">
                  <string>X</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="label_13">
                 <property name="text">
                  <string>Z</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_12">
                 <property name="text">
                  <string>Y</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="0">
                <widget class="QPushButton" name="decXButton">
                 <property name="text">
                  <string>Rotate X-</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QDoubleSpinBox" name="xField">
                 <property name="minimum">
                  <double>-5.000000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>5.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.100000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="6" column="1">
                <widget class="QPushButton" name="incYButton">
                 <property name="text">
                  <string>Rotate &amp;+</string>
                 </property>
                </widget>
               </item>
               <item row="7" column="1">
                <widget class="QPushButton" name="incZButton">
                 <property name="text">
                  <string>Rotate Z+</string>
                 </property>
                </widget>
               </item>
               <item row="7" column="0">
                <widget class="QPushButton" name="decZButton">
                 <property name="text">
                  <string>Rotate Z-</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="settingsWidget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QCheckBox" name="showMeshButton">
        <property name="text">
         <string>Show Mesh</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="showHullButton">
        <property name="text">
         <string>Show Bounding Hull</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="showSphereButton">
        <property name="text">
         <string>Show Bounding Sphere</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SceneWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>scenewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
